Bottom: 7125f7f07015d1cbe41b6ec0158eb62b0a81507d
Top:    09b860c62c6f4c145795a52cfd2c926c17429805
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-10-22 01:24:25 +0100

Refresh of try-something-with-the-rtrees

---

diff --git a/src/polygon1.c b/src/polygon1.c
index 1faf196..d9f96f4 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1626,11 +1626,7 @@ M_B_AREA_Collect (jmp_buf * e, POLYAREA * bfst, POLYAREA ** contours,
 		next = cur;
 		tmp->next = NULL;
 		tmp->Flags.status = UNKNWN;
-		/* Strictly we should pass b as the "owner" parameter, but
-		 * since the b POLYAREA is going to be delete anyway, it
-		 * doesn't matter that its contour r-tree gets out of sync.
-		 */
-		PutContour (e, tmp, contours, holes, NULL /* b */, NULL, NULL);
+		PutContour (e, tmp, contours, holes, b, NULL, NULL);
 		break;
 	      case PBO_UNITE:
 		break;		/* nothing to do - already included */
@@ -1646,11 +1642,7 @@ M_B_AREA_Collect (jmp_buf * e, POLYAREA * bfst, POLYAREA ** contours,
 		next = cur;
 		tmp->next = NULL;
 		tmp->Flags.status = UNKNWN;
-		/* Strictly we should pass b as the "owner" parameter, but
-		 * since the b POLYAREA is going to be delete anyway, it
-		 * doesn't matter that its contour r-tree gets out of sync.
-		 */
-		PutContour (e, tmp, contours, holes, NULL /* b */, NULL, NULL);
+		PutContour (e, tmp, contours, holes, b, NULL, NULL);
 		break;
 	      case PBO_ISECT:
 	      case PBO_SUB:
